backend/functions/utils.py

```diff
@@ -0,0 +1,152 @@
+import requests
+from users.utils import LANG_NAME_TO_CODE_ULCA, LANG_TRANS_MODEL_CODES
+
+### Utility Functions
+def get_batch_translations_using_indictrans_nmt_api(
+    sentence_list, source_language, target_language
+):
+
+    """Function to get the translation for the input sentences using the IndicTrans NMT API.
+
+    Args:
+        input_sentence (str): Sentence to be translated.
+        source_language (str): Original language of the sentence.
+        target_language (str): Final language of the sentence.
+
+    Returns:
+        list: List of dictionaries containing the translated sentences. 
+    """
+
+    # Get the translation model ID
+    model_id = LANG_TRANS_MODEL_CODES.get(f"{source_language}-{target_language}", 144)
+
+    # Convert language names to the language code
+    source_language = LANG_NAME_TO_CODE_ULCA[source_language]
+    target_language = LANG_NAME_TO_CODE_ULCA[target_language]
+
+    # Create the input sentences list
+    input_sentences = [{"source": sentence} for sentence in sentence_list]
+
+    headers = {
+        # Already added when you pass json= but not when you pass data=
+        # 'Content-Type': 'application/json',
+    }
+
+    json_data = {
+        "input": input_sentences,
+        "config": {
+            "modelId": model_id,
+            "language": {
+                "sourceLanguage": source_language,
+                "targetLanguage": target_language,
+            },
+        },
+    }
+
+    try:
+        response = requests.post(
+            "https://nmt-models.ulcacontrib.org/aai4b-nmt-inference/v0/translate",
+            headers=headers,
+            json=json_data,
+        )
+
+        output = ast.literal_eval(response.content.decode("utf-8"))

```

Comment:
How is this working without `import ast` ?

Also, `output = response.json()` should be sufficient right?

Reply:
Changed this to `response.json()`